src/md4c.c

```diff
@@ -3190,9 +3190,26 @@ md_collect_marks(MD_CTX* ctx, const MD_LINE* lines, int n_lines, int table_mode)
                 continue;
             }
 
+            /* A potential mention link. */
             /* A potential permissive e-mail autolink. */
             if(ch == _T('@')) {
-                if(line->beg + 1 <= off  &&  ISALNUM(off-1)  &&
+                if( (ctx->parser.flags & MD_FLAG_MENTIONS) && (line->beg == off || (CH(off-1) == _T(' '))) )
+                {
+                    OFF index = off + 1;
+                    if (index == line->end || CH(index) == ' ') {
+                        off++;
+                        continue;
+                    }
+                    while (index <= line->end)
+                    {
+                        if (!(ISALNUM(index) || (CH(index) == '_')))
+                            break;
+                        index++;
+                    }
+                    PUSH_MARK('@', off, index, MD_MARK_RESOLVED);
+                    off = index;
+                }
+                else if(line->beg + 1 <= off  &&  ISALNUM(off-1)  &&
```

Comment:
This condition now should likely explicitly check whether are enabled `MD_FLAG_PERMISSIVEEMAILAUTOLINKS`. That was not needed previously because `@` was added into the `mark_char_map[]` in `md_build_mark_char_map()` if and only if that extension was enabled.